### ISSUE TITLE
B023 location is wrong when occurring inside format string 

### ISSUE THREAD

**Issue by layday** [charliermarsh/ruff#910]
A late-bound loop variable used inside a format string is reported on the first line of the document.

```py
# stuff

# more stuff

for i in range(3):
    lambda: f'{i}'
```

ruff 0.0.139 output:

```
Found 1 error(s).
-:1:2: B023 Function definition does not bind loop variable `i`
```

**Comment by charliermarsh** [comment]
Ugh, f-strings. Best language feature as a user, worst language feature for tooling!

**Comment by charliermarsh** [comment]
(Thanks for filing, will fix.)
